.github/workflows/maven.yml

```diff
@@ -0,0 +1,41 @@
+name: Java CI with Maven
+
+on: [push, pull_request]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK 11
+      uses: actions/setup-java@v1
+      with:
+        java-version: 11
+
+    - name: Cache local Maven repository
+      uses: actions/cache@v2
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+
+    - name: Build with Maven
+      run: mvn -B -V package -DskipTests=true -Dmaven.javadoc.skip=true
+
+    - name: Tests with Coveralls coverage report
+      env:
+        COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
+      run: mvn clean test jacoco:report coveralls:report -DrepoToken="$COVERALLS_REPO_TOKEN"
+
+    - name: Deploy to GitHub pages
```

Comment:
This should probably be a separate workflow, unless we want a strategy to preserve the number of workflow being run?

In which case it can then use `$default_branch` as mentioned above.
We probably need to hash out a strategy for how we are going to handle these cases.

Reply:
In this case, I opt for including it in one workflow to reduce the number of times it has to build. There is a condition on the step that will have it only run on PR to tamu-master.

Reply:
When using separate workflows, we could in theory manually trigger just a coveralls rerun or more likely a ghpages rebuild, without running the other workflow jobs that the more monolithic approach would entail.

Reply:
In theory?